kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue

```diff
@@ -275,81 +273,70 @@
        */
       fetchLessonSiblings() {
+        // Map the progress for this lesson to start with
+        this.fetchContentNodeProgress({ lesson_id: this.lessonId });
+
+        // Get the lesson and then assign its resources to this.viewResourcesContents
         LessonResource.fetchModel({ id: this.lessonId }).then(lesson => {
           ContentNodeResource.fetchCollection({
             getParams: {
               ids: lesson.resources.map(resource => resource.contentnode_id),
             },
           }).then(contentNodes => {
-            return this.progressPromise(contentNodes);
+            // Filter out this.content
+            this.viewResourcesContents = contentNodes.filter(n => n.id !== this.content.id);
           });
         });
       },
       /**
        * Prepares a list of content nodes which are children of this.content.parent without
-       * this.content. The prepared nodes have their progress values for the current user
-       * (if there is one) mapped onto each node.
+       * this.content and calls fetchContentNodeProgress when the user is logged in.
        *
-       * Largely borrowed from modules/topicsTree/handlers.js
+       * Then it will fetch the "next folder" - which is the next content for this.content that
+       * is a topic.
        *
        * @modifies this.viewResourcesContents - Sets it to the progress-mapped nodes
-       * @modifies this.nextFolder - Sets the value with this.content's parents next sibling folder
+       * @modifies this.nextContent - Sets the value with this.content's parents next sibling folder
        * if found
+       * @modifies useContentNodeProgress.contentNodeProgressMap (indirectly) if the user
+       * is logged in
        */
       fetchSiblings() {
-        // Guard against this in lesson context
-        if (this.lessonId) {
-          return;
-        }
-
         ContentNodeResource.fetchTree({
           id: this.content.parent,
           params: {
             include_coach_content:
               this.$store.getters.isAdmin ||
               this.$store.getters.isCoach ||
               this.$store.getters.isSuperuser,
+            depth: 1,
           },
         }).then(parent => {
           // Filter out this.content
-          const nodes = parent.children.results.filter(node => node.id !== this.content.id);
+          this.viewResourcesContents = parent.children.results.filter(
+            n => n.id !== this.content.id
+          );
 
-          /** A promise to get the nextFolder content - done always */
-          const nextFolderPromise = ContentNodeResource.fetchNextContent(parent.id, {
+          // Fetch and map the progress for the nodes if logged in
+          if (this.$store.getters.isUserLoggedIn) {
+            const getParams = {
+              ids: this.viewResourcesContents.map(node => node.id),
+            };
+            this.fetchContentNodeProgress(getParams);
```

Comment:
Can avoid the delayed load here still by using `fetchContentNodeTreeProgress`:
```
this.fetchContentNodeTreeProgress({
          id: this.content.parent,
          params: {
            include_coach_content:
              this.$store.getters.isAdmin ||
              this.$store.getters.isCoach ||
              this.$store.getters.isSuperuser,
            depth: 1,
          },
        })
```

Can then be invoked in parallel to the node fetching.